### ISSUE TITLE
Deactivate questions on survey editor

### ISSUE THREAD

**Issue by wayangalihpratama** [akvo/isco#268]
- Backend
  - Add deactivate column to the question table
  - Filter questions by `deactivate=False` before uploading to bucket
 
- Frontend
  - Create deactivate UI
  - Check dependency when deactivating question (show prompt when the question has dependency children)
  - Show dependency children on a question if any (same as akvo-react-form-editor)

**Comment by joycarpediem** [comment]
@navins94 , i do not see a deactivate setting under question setting in survey editor.

@wayangalihpratama  Also please make sure that the following use case is handled :
If all questions in a question group are deactivated, then the question group is not displayed

**Comment by navins94** [comment]
@joycarpediem Initially we added deactivate button next to delete icon. Now we are moving this next to required button inside setting. This will take a little time.

![Screenshot 2023-01-20 at 3 18 17 PM](https://user-images.githubusercontent.com/24510012/213665613-18098bf5-521b-4b53-8b1f-827f6df5b47e.png)

**Comment by navins94** [comment]
@navins94 Add new endpoint to deactivate the question instead of existing save endpoint

**Comment by joycarpediem** [comment]
blocked by #314

**Comment by navins94** [comment]
Looking into it

**Comment by joycarpediem** [comment]
https://www.loom.com/share/ebec8a13b3764d83a814433c7d081c1a

Deactivate does not deactivate the dependent questions . It should deactivate all questions in the dependency chain . Currently it only deactivated the question that was displayed in the hover popup

**Comment by joycarpediem** [comment]
yes . The base principle is that if parent is deactivated, the child should also get deactivated

**Comment by joycarpediem** [comment]
Example 
  A
  |
  +-- B
       |
       +-- C

So if we deactivate question A, then question B will also be deactivated which will then also deactivate Question C

**Comment by navins94** [comment]
Hi @joycarpediem,
While enabling the questions lets suppose I am enabling `Am i happy` which is dependant on `Do you eat chocolate`. So the question is should we also enable `Do you eat chocolate` if we enable `Am I happy` or just the `Am I happy`
